lib/plugin.js

```diff
@@ -73,8 +73,16 @@ function startPageTracking(ctx) {
 }
 
 export default function (ctx, inject) {
-  const initialized = {<%= (options.autoInit && options.id) ? ` '${options.id}': true ` : '' %>}
-  ctx.$gtm = process.client ? gtmClient(ctx, initialized) : gtmServer(ctx, initialized)
+  const runtimeConfig = (ctx.$config && ctx.$config.gtm) || {}
+  const autoInit = <%= options.autoInit %>
+  const id = '<%= options.id %>'
```

Comment:
Can we drop [this line](https://github.com/nuxt-community/gtm-module/blob/20adee77b1e175f965b272ccd31ee79b550cb3d2/lib/plugin.js#L2) then?

Reply:
I was intended to replace all options with the merged ones (modules + runtime), however, I noticed that some actions are taken previously on the module, which only runs in build.

This forces the ID used in build time, even if you replace on runtime, so I kept everything as is and only added an additional init with the runtime ID, if different from the defined on build.

Reply:
What should be the best option, in this case?

Reply:
Indeed makes sense to not support other options because of module logic. One idea for this gtm module could be supporting runtimeConfig from module and make it runtime module.

Reply:
This should be done in another issue/PR or may I propose this changes here?

_I'm kinda new on contributing to open source projects_ 🙈

Reply:
Another PR sounds good too. Glad having you onboard @SkyaTura thanks for PRs ❤️

Reply:
This changes are no longer needed, since the module itself will be runtime, we can grab the configurations before compiling the plugin.